packages/core/core/src/services/document-service/transform/filters.ts

```diff
@@ -0,0 +1,58 @@
+import { traverse } from '@strapi/utils';
+import { transformFields } from './fields';
+
+import { type Data, type Options } from './types';
+import { switchDocumentIdForId } from './utils';
+
+const relationalIdReplacement = (filter: Data, opts: Options) => {
+  // If there is an id field in the filter object, replace it with documentId
+  // and provide the filter parameters for the publishedAt and locale fields
+  if ('id' in filter) {
+    filter.documentId = filter.id;
+    delete filter.id;
+
+    filter.publishedAt = opts.isDraft ? null : { $ne: null };
+
+    if (opts.locale) {
+      filter.locale = opts.locale;
+    }
```

Comment:
I am sorry, was just testing this and realized this might not be necessary 🤦‍♂️ 
Relations are already connected to the matching dimensiosn (locale and publication state). Filter works but it's not acutally necessary.

Was checking, and it is only in cases where one of the sides have a feature disabled (e.g. i18n) where we would list all relations , but I think thats ok, and the user can always filter

Reply:
😊 no worries, so we actually only need to make the id switch?

Reply:
yep! I think so :)